kedro_kubeflow/auth.py

```diff
@@ -1,7 +1,13 @@
 import logging
 import os
+import re
+from urllib.parse import urlsplit, urlunsplit
+
+import requests
 
 IAP_CLIENT_ID = "IAP_CLIENT_ID"
+DEX_USERNAME = "DEX_USERNAME"
+DEX_PASSWORD = "DEX_PASSWORD"
 
 
 class AuthHandler(object):
```

Comment:
That looks fine for now, but moving forward I would imagine handling auth in a more generic way. E.g. we have `abstract` class AuthHandler that has required methods (obtain token) and in config we specify something like:
```
auth: 
    handler: kedro_kubeflow.auth.DexAuthHandler
    params: ....
```
where handler class gets params as dict to constructor whatever they are. Sth like that would allow to implement any kind of auth without modifying the plugin.

Reply:
That's fantastic idea! Let's follow this approach while adding some third authentication method.